app/controllers/facility_accounts_controller.rb

```diff
@@ -79,13 +73,12 @@ def edit
 
   # PUT /facilities/:facility_id/accounts/:id
   def update
-    class_params = account_class_params
-
     if @account.is_a?(AffiliateAccount)
-      class_params[:affiliate_other] = nil if class_params[:affiliate_id] != Affiliate.OTHER.id.to_s
+      @account.affiliate_other = nil if params[:affiliate_id] != Affiliate.OTHER.id.to_s
     end
```

Comment:
There's global a table of affiliates managed under "global settings", which is what `affiliate_id` is foreign keyed to. When creating a CC/PO, you can either select one of the existing affiliates from the dropdown, or choose "Other" (which has an entry in the affiliates table) and enter a free-text field (`affiliate_other`). If `affiliate_other is not null`, then `affiliate_id = 1` (as that's the id of "Other").